runtime/lua/vim/lsp/callbacks.lua

```diff
@@ -54,9 +54,10 @@ M['textDocument/references'] = function(_, _, result)
   api.nvim_command("wincmd p")
 end
 
-local symbol_callback = function(_, _, result, _, bufnr)
```

Comment:
This should not be removed.
See this commit message.
https://github.com/neovim/neovim/pull/11931/commits/f157fdef7eccba6601c6d3b1475847466b60c25b

I think that if bufnr is nil and uri scheme is not file,  use uri as fname.

Reply:
> This should not be removed.
> See this commit message.
> f157fde

Thx for the pointer, I'll have a closer look why bufnr was nil for me.



>  think that if bufnr is nil and uri scheme is not file, use uri as fname.

Is there any advantage to use `filename` instead of `bufnr` in the items?

Reply:
Shouldn't `documentSymbol` always be only for the _current_ buffer anyway (hence no location -- in contrast to `workspaceSymbol`)?

Reply:
@clason 
https://github.com/neovim/neovim/pull/12264#discussion_r421377417

Reply:
@mfussenegger 
> Is there any advantage to use filename instead of bufnr in the items?

I was misreading the code because I was looking at it from mobile.
I would like to investigate why bufnr becomes nil, so would you give me a minimal case to reproduce it?

Reply:
@h-michael My point was if it wasn't possible (and easier) to just fix `0` instead of `bufnr` in this callback? (honest question.)

Reply:
@mfussenegger @clason 
I'm so sorry. I misunderstood various things.

To avoid race conditions, I think it is better to investigate why `bufnr` becomes nil and fix it if possible.

I think we can use either `fname` or `bufnr` for qlist.
I thought that we should use `fname` because `workspace/symbol` does not always refer to a file that has already been opened, so we should use `fname`, but if the target file does not already have a buffer, `uri_to_bufnr` does `bufadd` so there should be no problem.